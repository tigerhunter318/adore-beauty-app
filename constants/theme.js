import { Platform, PlatformColor } from 'react-native'

const theme = {
  black: '#000000',
  black10: 'rgba(0,0,0,0.1)',
  black20: 'rgba(0,0,0,0.2)',
  black30: 'rgba(0,0,0,0.3)',
  black40: 'rgba(0,0,0,0.4)',
  black50: 'rgba(0,0,0,0.5)',
  black60: 'rgba(0,0,0,0.6)',
  black70: 'rgba(0,0,0,0.7)',
  black80: 'rgba(0,0,0,0.8)',
  black90: 'rgba(0,0,0,0.9)',
  black100: 'rgba(0,0,0,1)',
  white: '#FFFFFF',
  green: '#4f8a0f',
  iOSKeyboard: '#d1d3d9',
  textGrey: '#BCBCBC',
  textGreyDark: '#777777',
  textMediumGrey: '#999999',
  borderColorDark: '#ACACAC',
  borderColor: '#E1DFDF',
  reviewBorderColor: '#EBEBEB',
  lightYellow: '#FBECDF',
  backgroundLightGrey: 'rgba(221,221,221,0.2)',
  lightGrey: '#f5f3f3',
  backgroundLighterGrey: 'rgba(225,223,223,0.41)',
  lightBlue: '#82B5C9',
  lightBackgroundBlue: '#f0f3f4',
  peach: '#FFA79D',
  darkBlack: '#212B36',
  lightBlack: '#404040',
  lighterBlack: '#4A4A4A',
  darkGray: '#979797',
  darkGrayWithOpacity: 'rgba(151, 151, 151, 0.2)',
  tabColor: 'rgba(102, 102, 102, 0.1)',
  pinkColor: '#E93777',
  titleBorderColor: '#201f26',
  red: '#FF0000',
  darkRed: '#cc0000',
  lighterRed: '#FFE0DC',
  lightPink: '#FBECDF',
  pink: '#FFCEC8',
  lighterPeach: '#fcf6ef',
  lighterPink: 'rgba(251, 236, 223, 0.5)',
  splitorColor: '#030303',
  orange: '#D6775E',
  orangeWithOpacity: 'rgba(214, 119, 94, 0.7)',
  darkerGray: 'rgba(36, 36, 36, 0.07)',
  lightBorder: '#CCCCCC',
  orangeColor: '#FFAF54',
  lightOrange: '#FECF96',
  codeOrange: '#FBECDE',
  lightPeach: '#FFCACB',
  backgroundGrey: 'rgba(242,242,242,1)',
  tabTextColor: 'rgba(74, 74, 74, 0.5)',
  darkOrange: '#F7C890',
  darkPeach: '#E3BDB1',
  tabBackgroundColor: 'rgba(216, 216, 216, 0.22)',
  level1: '#DB846D',
  level2: '#9DA483',
  level3: '#666666',
  giftIconLevel1: '#D6775E',
  giftIconLevel2: '#8BAC6C',
  giftIconLevel3: '#000000',
  stars: '#f7c868',
  linkColor: Platform.select({
    ios: PlatformColor('link'),
    android: PlatformColor('@android:color/holo_blue_light'),
    default: '#147EFB'
  })
}

export default theme
