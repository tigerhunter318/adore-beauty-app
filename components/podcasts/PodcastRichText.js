import React, { useState } from 'react'
import HTMLView from 'react-native-htmlview'
import { StyleSheet } from 'react-native'
import Container from '../ui/Container'
import Type from '../ui/Type'
import theme from '../../constants/theme'
import { sanitizeContent, trimHtmlSpaces } from '../../utils/format'
import { usePodcastPlayerContext } from './PodcastPlayerContext'
import { navigateUrlToScreen } from '../../navigation/utils/useUrlNavigation'
import resolveUrlToContent from '../../navigation/utils/resolveUrlToContent'
import { urlRegex } from '../../utils/validation'
import LoadingOverlay from '../ui/LoadingOverlay'

const styles = StyleSheet.create({
  p: {
    fontSize: 16,
    lineHeight: 24,
    color: theme.lightBlack,
    letterSpacing: 0.1,
    paddingHorizontal: 20,
    marginBottom: 10,
    marginTop: 10
  },
  a: {
    textDecorationLine: 'underline'
  }
})

const PodcastRichText = ({
  content,
  color = theme.white,
  center = false,
  hasBoldLinks = false,
  styleProps = {},
  isModalScreen = false,
  ignoreLinks = false,
  ...rest
}) => {
  const [isLoading, setIsLoading] = useState(false)
  const { bottomSheetRef, navigation, minimizePlayer } = usePodcastPlayerContext()

  const handleAudioPlayerMinimize = () => {
    if (bottomSheetRef?.current?.isOpen) {
      minimizePlayer()
    }
  }

  const handleRedirect = async url => {
    setIsLoading(true)

    if (isModalScreen) {
      handleAudioPlayerMinimize()
    }

    // TODO: Resolve navigation issue with CMS

    const response = await resolveUrlToContent(url)
    if (response?.type === 'cms') {
      navigation.navigate('CMS', { ...response })
      setIsLoading(false)
      return
    }

    await navigateUrlToScreen(navigation, url, 'navigate')
    setIsLoading(false)
  }

  const renderNode = (node, index) => {
    if (!node) return

    const { name, children, attribs, type, data } = node

    const childrenData = children?.[0]?.data

    if (name === 'p' && urlRegex.test(childrenData)) {
      const text = childrenData.split('http')?.[0]
      const link = childrenData.match(urlRegex)?.[0]

      return (
        <Type
          ph={2}
          key={`a-href-${index}`}
          style={[
            styles.p,
            {
              paddingHorizontal: 0,
              textAlign: center ? 'center' : 'left',
              color
            },
            styleProps?.p
          ]}
        >
          {text}{' '}
          <Type
            semiBold={hasBoldLinks}
            underline
            onPress={() => handleRedirect(link)}
            color={color}
            letterSpacing={0.1}
          >
            {link}
          </Type>
        </Type>
      )
    }

    if (name === 'strong' && childrenData) {
      return (
        <Type key={`strong-${index}`}>
          {node?.prev && ' '}
          <Type color={color} bold>
            {trimHtmlSpaces(childrenData)}
          </Type>
          {node?.next && ' '}
        </Type>
      )
    }

    if (name === 'a' && attribs.href) {
      if (childrenData) {
        const tag = childrenData.split('?')?.[0]

        if (!ignoreLinks) {
          return (
            <Type key={`a-href-${index}`}>
              {node?.prev && ' '}
              <Type
                semiBold={hasBoldLinks}
                underline
                color={color}
                letterSpacing={0.1}
                onPress={() => handleRedirect(attribs.href)}
              >
                {trimHtmlSpaces(tag)}
              </Type>
              {node?.next && ' '}
            </Type>
          )
        }

        return (
          <Type key={`a-href-${index}`}>
            {node?.prev && ' '}
            <Type
              style={[
                styles.p,
                {
                  paddingHorizontal: 0,
                  textAlign: center ? 'center' : 'left',
                  color
                },
                styleProps?.p
              ]}
              onPress={() => handleRedirect(attribs.href)}
            >
              {trimHtmlSpaces(tag)}
            </Type>
            {node?.next && ' '}
          </Type>
        )
      }
    }

    if (data && type === 'text' && data.trim().length > 0) {
      return (
        <Type
          key={`none-tag-content-${index}`}
          style={[
            styles.p,
            {
              paddingHorizontal: 0,
              textAlign: center ? 'center' : 'left',
              color
            },
            styleProps?.p
          ]}
        >
          {trimHtmlSpaces(data)}
        </Type>
      )
    }
  }

  return (
    <Container style={styleProps.container}>
      <HTMLView
        value={sanitizeContent(content)}
        addLineBreaks={false}
        stylesheet={styles}
        renderNode={renderNode}
        {...rest}
      />
      <LoadingOverlay active={isLoading} />
    </Container>
  )
}

export default PodcastRichText
