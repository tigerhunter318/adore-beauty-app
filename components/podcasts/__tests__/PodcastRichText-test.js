import React from 'react'
import renderer from 'react-test-renderer'
import theme from '../../../constants/theme'
import PodcastRichText from '../PodcastRichText'

/* eslint-disable global-require */

jest.mock('../PodcastPlayerContext.tsx', () => ({
  usePodcastPlayerContext: () => ({})
}))

it(`PodcastRichText renders for a program`, () => {
  const mockProgram = require('./__mocks__/beauty-iq-uncensored.json')
  const { DescriptionHtml: content } = mockProgram || {}
  const tree = renderer.create(<PodcastRichText ignoreLinks content={content} color={theme.lightBlack} />).toJSON()

  expect(tree).toMatchSnapshot()
})

it(`PodcastRichText renders for an episode`, () => {
  const mockEpisode = require('./__mocks__/ep-135-everything-you-need-to-know-about-rosacea.json')
  const { content } = mockEpisode
  const tree = renderer
    .create(<PodcastRichText content={content} color={theme.black} styleProps={{ p: { color: theme.lightBlack } }} />)
    .toJSON()

  expect(tree).toMatchSnapshot()
})
