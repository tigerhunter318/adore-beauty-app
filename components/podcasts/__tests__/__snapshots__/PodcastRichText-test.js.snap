// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PodcastRichText renders for a program 1`] = `
<View
  style={
    Array [
      Object {},
      Object {},
    ]
  }
>
  <View>
    <Text
      onLongPress={null}
      onPress={null}
      style={
        Object {
          "color": "#404040",
          "fontSize": 16,
          "letterSpacing": 0.1,
          "lineHeight": 24,
          "marginBottom": 10,
          "marginTop": 10,
          "paddingHorizontal": 20,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "fontFamily": "Montserrat-Regular",
            },
            Array [
              Object {
                "color": "#404040",
                "fontSize": 16,
                "letterSpacing": 0.1,
                "lineHeight": 24,
                "marginBottom": 10,
                "marginTop": 10,
                "paddingHorizontal": 20,
              },
              Object {
                "color": "#404040",
                "paddingHorizontal": 0,
                "textAlign": "left",
              },
              undefined,
            ],
          ]
        }
      >
        Introducing
      </Text>
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "fontFamily": "Montserrat-Regular",
            },
            undefined,
          ]
        }
      >
         
        <Text
          allowFontScaling={false}
          onPress={[Function]}
          style={
            Array [
              Object {
                "fontFamily": "Montserrat-Regular",
              },
              Array [
                Object {
                  "color": "#404040",
                  "fontSize": 16,
                  "letterSpacing": 0.1,
                  "lineHeight": 24,
                  "marginBottom": 10,
                  "marginTop": 10,
                  "paddingHorizontal": 20,
                },
                Object {
                  "color": "#404040",
                  "paddingHorizontal": 0,
                  "textAlign": "left",
                },
                undefined,
              ],
            ]
          }
        >
          Adore Beauty
        </Text>
         
      </Text>
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "fontFamily": "Montserrat-Regular",
            },
            Array [
              Object {
                "color": "#404040",
                "fontSize": 16,
                "letterSpacing": 0.1,
                "lineHeight": 24,
                "marginBottom": 10,
                "marginTop": 10,
                "paddingHorizontal": 20,
              },
              Object {
                "color": "#404040",
                "paddingHorizontal": 0,
                "textAlign": "left",
              },
              undefined,
            ],
          ]
        }
      >
        's new podcast Beauty IQ Uncensored, with hosts Beauty Editor Joanna and beauty newbie Hannah getting real about every aspect of beauty. No topic is off-limits - from retinol to bum hair - and you can expect a healthy dose of education and laughs. Warning: Not suitable for the school pick up.
      </Text>
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "fontFamily": "Montserrat-Regular",
            },
            undefined,
          ]
        }
      >
         
        <Text
          allowFontScaling={false}
          onPress={[Function]}
          style={
            Array [
              Object {
                "fontFamily": "Montserrat-Regular",
              },
              Array [
                Object {
                  "color": "#404040",
                  "fontSize": 16,
                  "letterSpacing": 0.1,
                  "lineHeight": 24,
                  "marginBottom": 10,
                  "marginTop": 10,
                  "paddingHorizontal": 20,
                },
                Object {
                  "color": "#404040",
                  "paddingHorizontal": 0,
                  "textAlign": "left",
                },
                undefined,
              ],
            ]
          }
        >
          www.adorebeauty.com.au
        </Text>
         
      </Text>
      <Text
        style={
          Array [
            null,
            Object {
              "color": "#404040",
              "fontSize": 16,
              "letterSpacing": 0.1,
              "lineHeight": 24,
              "marginBottom": 10,
              "marginTop": 10,
              "paddingHorizontal": 20,
            },
          ]
        }
      >
         
      </Text>
    </Text>
  </View>
</View>
`;

exports[`PodcastRichText renders for an episode 1`] = `
<View
  style={
    Array [
      Object {},
      Object {},
    ]
  }
>
  <View>
    <Text
      onLongPress={null}
      onPress={null}
      style={
        Object {
          "color": "#404040",
          "fontSize": 16,
          "letterSpacing": 0.1,
          "lineHeight": 24,
          "marginBottom": 10,
          "marginTop": 10,
          "paddingHorizontal": 20,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "fontFamily": "Montserrat-Regular",
            },
            undefined,
          ]
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "Montserrat-Bold",
              },
              undefined,
            ]
          }
        >
          Does The Squatty Potty Actually Work?
        </Text>
      </Text>
    </Text>
    <Text
      onLongPress={null}
      onPress={null}
      style={
        Object {
          "color": "#404040",
          "fontSize": 16,
          "letterSpacing": 0.1,
          "lineHeight": 24,
          "marginBottom": 10,
          "marginTop": 10,
          "paddingHorizontal": 20,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "fontFamily": "Montserrat-Regular",
            },
            Array [
              Object {
                "color": "#404040",
                "fontSize": 16,
                "letterSpacing": 0.1,
                "lineHeight": 24,
                "marginBottom": 10,
                "marginTop": 10,
                "paddingHorizontal": 20,
              },
              Object {
                "color": "#000000",
                "paddingHorizontal": 0,
                "textAlign": "left",
              },
              Object {
                "color": "#404040",
              },
            ],
          ]
        }
      >
        In true fashion, our resident GP Dr Lucinda is back on the podcast to tackle this topic. Along with discussing the specifics of the squatty potty, Dr Lucinda also schools us on the best poo techniques.
      </Text>
    </Text>
    <Text
      onLongPress={null}
      onPress={null}
      style={
        Object {
          "color": "#404040",
          "fontSize": 16,
          "letterSpacing": 0.1,
          "lineHeight": 24,
          "marginBottom": 10,
          "marginTop": 10,
          "paddingHorizontal": 20,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "fontFamily": "Montserrat-Regular",
            },
            Array [
              Object {
                "color": "#404040",
                "fontSize": 16,
                "letterSpacing": 0.1,
                "lineHeight": 24,
                "marginBottom": 10,
                "marginTop": 10,
                "paddingHorizontal": 20,
              },
              Object {
                "color": "#000000",
                "paddingHorizontal": 0,
                "textAlign": "left",
              },
              Object {
                "color": "#404040",
              },
            ],
          ]
        }
      >
        Find Dr Lucinda on Instagram here:
      </Text>
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "fontFamily": "Montserrat-Regular",
            },
            undefined,
          ]
        }
      >
         
        <Text
          allowFontScaling={false}
          onPress={[Function]}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "Montserrat-Regular",
                "letterSpacing": 0.1,
                "textDecorationLine": "underline",
              },
              undefined,
            ]
          }
        >
          https://www.instagram.com/todaywithdrlucia/
        </Text>
      </Text>
    </Text>
    <Text
      onLongPress={null}
      onPress={null}
      style={
        Object {
          "color": "#404040",
          "fontSize": 16,
          "letterSpacing": 0.1,
          "lineHeight": 24,
          "marginBottom": 10,
          "marginTop": 10,
          "paddingHorizontal": 20,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "fontFamily": "Montserrat-Regular",
            },
            undefined,
          ]
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "Montserrat-Bold",
              },
              undefined,
            ]
          }
        >
          Rosacea
        </Text>
      </Text>
    </Text>
    <Text
      onLongPress={null}
      onPress={null}
      style={
        Object {
          "color": "#404040",
          "fontSize": 16,
          "letterSpacing": 0.1,
          "lineHeight": 24,
          "marginBottom": 10,
          "marginTop": 10,
          "paddingHorizontal": 20,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "fontFamily": "Montserrat-Regular",
            },
            Array [
              Object {
                "color": "#404040",
                "fontSize": 16,
                "letterSpacing": 0.1,
                "lineHeight": 24,
                "marginBottom": 10,
                "marginTop": 10,
                "paddingHorizontal": 20,
              },
              Object {
                "color": "#000000",
                "paddingHorizontal": 0,
                "textAlign": "left",
              },
              Object {
                "color": "#404040",
              },
            ],
          ]
        }
      >
        Dr Cara McDonald joins us for our first ever segment dedicated to rosacea (how has it taken us this long?)
      </Text>
    </Text>
    <Text
      onLongPress={null}
      onPress={null}
      style={
        Object {
          "color": "#404040",
          "fontSize": 16,
          "letterSpacing": 0.1,
          "lineHeight": 24,
          "marginBottom": 10,
          "marginTop": 10,
          "paddingHorizontal": 20,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "fontFamily": "Montserrat-Regular",
            },
            Array [
              Object {
                "color": "#404040",
                "fontSize": 16,
                "letterSpacing": 0.1,
                "lineHeight": 24,
                "marginBottom": 10,
                "marginTop": 10,
                "paddingHorizontal": 20,
              },
              Object {
                "color": "#000000",
                "paddingHorizontal": 0,
                "textAlign": "left",
              },
              Object {
                "color": "#404040",
              },
            ],
          ]
        }
      >
        You can learn more about Dr Cara McDonald here:
      </Text>
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "fontFamily": "Montserrat-Regular",
            },
            undefined,
          ]
        }
      >
         
        <Text
          allowFontScaling={false}
          onPress={[Function]}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "Montserrat-Regular",
                "letterSpacing": 0.1,
                "textDecorationLine": "underline",
              },
              undefined,
            ]
          }
        >
          https://completeskinspecialists.com.au/our-team/dr-cara-mcdonald-dermatologist/
        </Text>
      </Text>
    </Text>
    <Text
      onLongPress={null}
      onPress={null}
      style={
        Object {
          "color": "#404040",
          "fontSize": 16,
          "letterSpacing": 0.1,
          "lineHeight": 24,
          "marginBottom": 10,
          "marginTop": 10,
          "paddingHorizontal": 20,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "fontFamily": "Montserrat-Regular",
            },
            undefined,
          ]
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "Montserrat-Bold",
              },
              undefined,
            ]
          }
        >
          PWDKWN:
        </Text>
      </Text>
    </Text>
    <Text
      onLongPress={null}
      onPress={null}
      style={
        Object {
          "color": "#404040",
          "fontSize": 16,
          "letterSpacing": 0.1,
          "lineHeight": 24,
          "marginBottom": 10,
          "marginTop": 10,
          "paddingHorizontal": 20,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "fontFamily": "Montserrat-Regular",
            },
            Array [
              Object {
                "color": "#404040",
                "fontSize": 16,
                "letterSpacing": 0.1,
                "lineHeight": 24,
                "marginBottom": 10,
                "marginTop": 10,
                "paddingHorizontal": 20,
              },
              Object {
                "color": "#000000",
                "paddingHorizontal": 0,
                "textAlign": "left",
              },
              Object {
                "color": "#404040",
              },
            ],
          ]
        }
      >
        Jo: Olaplex No. 9 Bond Protector Nourishing Hair Serum
      </Text>
    </Text>
    <Text
      onLongPress={null}
      onPress={null}
      style={
        Object {
          "color": "#404040",
          "fontSize": 16,
          "letterSpacing": 0.1,
          "lineHeight": 24,
          "marginBottom": 10,
          "marginTop": 10,
          "paddingHorizontal": 20,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "fontFamily": "Montserrat-Regular",
            },
            undefined,
          ]
        }
      >
        <Text
          allowFontScaling={false}
          onPress={[Function]}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "Montserrat-Regular",
                "letterSpacing": 0.1,
                "textDecorationLine": "underline",
              },
              undefined,
            ]
          }
        >
          https://www.adorebeauty.com.au/olaplex/olaplex-no-9-bond-protector-nourishing-hair-serum.html
        </Text>
      </Text>
    </Text>
    <Text
      onLongPress={null}
      onPress={null}
      style={
        Object {
          "color": "#404040",
          "fontSize": 16,
          "letterSpacing": 0.1,
          "lineHeight": 24,
          "marginBottom": 10,
          "marginTop": 10,
          "paddingHorizontal": 20,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "fontFamily": "Montserrat-Regular",
            },
            Array [
              Object {
                "color": "#404040",
                "fontSize": 16,
                "letterSpacing": 0.1,
                "lineHeight": 24,
                "marginBottom": 10,
                "marginTop": 10,
                "paddingHorizontal": 20,
              },
              Object {
                "color": "#000000",
                "paddingHorizontal": 0,
                "textAlign": "left",
              },
              Object {
                "color": "#404040",
              },
            ],
          ]
        }
      >
        Hannah: Alpha-H Generation Glow Daily Resurfacing Essence
      </Text>
    </Text>
    <Text
      onLongPress={null}
      onPress={null}
      style={
        Object {
          "color": "#404040",
          "fontSize": 16,
          "letterSpacing": 0.1,
          "lineHeight": 24,
          "marginBottom": 10,
          "marginTop": 10,
          "paddingHorizontal": 20,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "fontFamily": "Montserrat-Regular",
            },
            undefined,
          ]
        }
      >
        <Text
          allowFontScaling={false}
          onPress={[Function]}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "Montserrat-Regular",
                "letterSpacing": 0.1,
                "textDecorationLine": "underline",
              },
              undefined,
            ]
          }
        >
          https://www.adorebeauty.com.au/alpha-h/alpha-h-generation-glow-daily-resurfacing-essence.html
        </Text>
      </Text>
    </Text>
    <Text
      onLongPress={null}
      onPress={null}
      style={
        Object {
          "color": "#404040",
          "fontSize": 16,
          "letterSpacing": 0.1,
          "lineHeight": 24,
          "marginBottom": 10,
          "marginTop": 10,
          "paddingHorizontal": 20,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "fontFamily": "Montserrat-Regular",
            },
            Array [
              Object {
                "color": "#404040",
                "fontSize": 16,
                "letterSpacing": 0.1,
                "lineHeight": 24,
                "marginBottom": 10,
                "marginTop": 10,
                "paddingHorizontal": 20,
              },
              Object {
                "color": "#000000",
                "paddingHorizontal": 0,
                "textAlign": "left",
              },
              Object {
                "color": "#404040",
              },
            ],
          ]
        }
      >
        We have a Facebook Group! Join the Beauty IQ Uncensored FB group to share beauty tips, chat about the podcast and share your own cringey stories:
      </Text>
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "fontFamily": "Montserrat-Regular",
            },
            undefined,
          ]
        }
      >
         
        <Text
          allowFontScaling={false}
          onPress={[Function]}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "Montserrat-Regular",
                "letterSpacing": 0.1,
                "textDecorationLine": "underline",
              },
              undefined,
            ]
          }
        >
          https://www.facebook.com/groups/484267299748882/
        </Text>
      </Text>
    </Text>
    <Text
      allowFontScaling={false}
      ph={2}
      style={
        Array [
          Object {
            "fontFamily": "Montserrat-Regular",
            "paddingLeft": 20,
            "paddingRight": 20,
          },
          Array [
            Object {
              "color": "#404040",
              "fontSize": 16,
              "letterSpacing": 0.1,
              "lineHeight": 24,
              "marginBottom": 10,
              "marginTop": 10,
              "paddingHorizontal": 20,
            },
            Object {
              "color": "#000000",
              "paddingHorizontal": 0,
              "textAlign": "left",
            },
            Object {
              "color": "#404040",
            },
          ],
        ]
      }
    >
      Disclaimer: 
       
      <Text
        allowFontScaling={false}
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "Montserrat-Regular",
              "letterSpacing": 0.1,
              "textDecorationLine": "underline",
            },
            undefined,
          ]
        }
      >
        https://www.adorebeauty.com.au/disclaimer.html
      </Text>
    </Text>
    <Text
      onLongPress={null}
      onPress={null}
      style={
        Object {
          "color": "#404040",
          "fontSize": 16,
          "letterSpacing": 0.1,
          "lineHeight": 24,
          "marginBottom": 10,
          "marginTop": 10,
          "paddingHorizontal": 20,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "fontFamily": "Montserrat-Regular",
            },
            Array [
              Object {
                "color": "#404040",
                "fontSize": 16,
                "letterSpacing": 0.1,
                "lineHeight": 24,
                "marginBottom": 10,
                "marginTop": 10,
                "paddingHorizontal": 20,
              },
              Object {
                "color": "#000000",
                "paddingHorizontal": 0,
                "textAlign": "left",
              },
              Object {
                "color": "#404040",
              },
            ],
          ]
        }
      >
        Hosts: Joanna Fleming & Hannah Furst
      </Text>
    </Text>
    <Text
      onLongPress={null}
      onPress={null}
      style={
        Object {
          "color": "#404040",
          "fontSize": 16,
          "letterSpacing": 0.1,
          "lineHeight": 24,
          "marginBottom": 10,
          "marginTop": 10,
          "paddingHorizontal": 20,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "fontFamily": "Montserrat-Regular",
            },
            Array [
              Object {
                "color": "#404040",
                "fontSize": 16,
                "letterSpacing": 0.1,
                "lineHeight": 24,
                "marginBottom": 10,
                "marginTop": 10,
                "paddingHorizontal": 20,
              },
              Object {
                "color": "#000000",
                "paddingHorizontal": 0,
                "textAlign": "left",
              },
              Object {
                "color": "#404040",
              },
            ],
          ]
        }
      >
        Guests: Dr Lucinda & Dr Cara McDonald
      </Text>
    </Text>
  </View>
</View>
`;
